lib/server/swarm.js

```diff
@@ -98,15 +108,19 @@ Swarm.prototype._onAnnounceUpdate = function (params, peer) {
     this.complete += 1
     this.incomplete -= 1
     peer.complete = true
+    var id = params.type === 'ws' ? params.peer_id : params.addr
+    this.peers.set(id, peer)
```

Comment:
Can we change this to a `get()` and pass in `id` as a parameter? All 4 announce event types now require the id, so that would be cleaner.